src/Rest/Helpers/Command/CommandBuilder.php

```diff
@@ -194,4 +195,59 @@ public function get(): array
 
         return $data;
     }
+
+    public function matchesApplicationCommand(ApplicationCommand $applicationCommand): bool
+    {
+        $applicationCommandBuilder = self::fromApplicationCommand($applicationCommand)->get();
+
+        foreach ($this->get() as $key => $value) {
+            if ($applicationCommandBuilder[$key] !== $value) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    protected static function fromApplicationCommand(ApplicationCommand $applicationCommand): self
```

Comment:
Another very fun thing to keep in mind for this function, guild commands don't have `default_member_permissions`, but they may still be present in the builder as I didn't see a need to have 2 separate builders just for this property. Also because of the dynamic `registerCommand` functionality

Reply:
Global commands registered in dev mode are converted to guild commands. The builder will have it filled it, the response from discord will not have it.